### DISCUSSION TITLE
Turbopack msgpackr package error in require optional packages

### DISCUSSION THREAD

**Discussion by joacub** [vercel/next.js#62588] [Help]
### Link to the code that reproduces this issue

https://codesandbox.io/p/github/avinashcodelabs/nextjs-turbo-socketio-reproduce-issue/csb-jytnxy/draft/ancient-fire?file=%2Fpackage.json%3A12%2C27

### To Reproduce

just start the app and will get the error

### Current vs. Expected behavior

not building

### Provide environment information

```bash
last canary version
```

### Which area(s) are affected? (Select all that apply)

App Router, Turbopack (--turbo)

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

msgpackr package not compatible with turbo

**Comment by balazsorban44** [comment]
I was getting the same behavior on both `next dev --turbo` and `next dev`, but could also not find `msgpackr` as part of the reproduction. Also, the lockfile indicates that you were not using `next@canary`, but hard to tell since the `next info` output was not provided. Make sure to test with the latest canary and re-open the issue with a reproduction if it still happens. Don't forget to add `msgpackr` if it is the erroneous package.

**Reply by joacub** [reply]
its seems ti be other package, msgpackr-extract that is linked to this one

**Comment by joacub** [comment]
the error comes in a conditionally with try catch of a require of a optional package in the package msgpackr

this is the code causing turbopack to fail:

```js
if (!nativeAccelerationDisabled) {
	let extractor;
	try {
		if (typeof require == 'function')
			extractor = require('msgpackr-extract');
		else
			extractor = module$1.createRequire((typeof document === 'undefined' ? new (require('u' + 'rl').URL)('file:' + __filename).href : (document.currentScript && document.currentScript.src || new URL('node.cjs', document.baseURI).href)))('msgpackr-extract');
		if (extractor)
			setExtractor(extractor.extractStrings);
	} catch (error) {
		// native module is optional
	}
}
```
